### PR TITLE
REST API Monitoring

### LINKED ISSUE #3545: REST API: Allow user to check/manage Monitoring of API
**Title**:
REST API: Allow user to check/manage Monitoring of API

**Description**:

TBA

**User Story**:

"As an API owner, I need to enable Monitoring of my API."
"As an API owner, I need to manage Monitoring of my API."
"As an developer owner, I need to check status of API."

### PR DESCRIPTION
**Pull request by 55**
Closes #3545

This brings BASIC functionality of API Monitoring.

## Changes
- A new GET method to show latest monitoring status code(s)

## Developer checklist
- [x] Alternative solutions were compared/discussed before writing code
    - [x] trade-offs with this solution are considered acceptable
- [x] Code in this PR adheres to the [project styleguide](CONTRIBUTING.md)
- [x] This pull request does not decrease project test coverage
- [x] If the code changes existing database collection(s), migration has been written
- [x] If UI texts are added or changed, all texts are internationalized

## Reviewer checklist
Reviewed by: 

- [ ] Code works as described/expected
- [ ] Code seems to be error free
    - [ ] no browser console errors visible
    - [ ] no server console errors visible
    - [ ] passes CI build
- [ ] Code is written in a way that promotes maintainability
    - [ ] easy to understand
    - [ ] well organized
    - [ ] follows project coding standards and conventions
    - [ ] well documented

### CODE REVIEW

File: apinf_packages/apis/server/api.js [hunk 1]
```diff
@@ -1921,5 +1923,120 @@ CatalogV1.addRoute('apis/:id/proxyBackend', {
...
+  get: {
+    authRequired: false,
+    swagger: {
+      tags: [
+        CatalogV1.swagger.tags.api,
+      ],
+      summary: 'Get latest Monitoring status of API',
+      description: 'In return user get object with latest API code status',
+      parameters: [
+        CatalogV1.swagger.params.apiId,
```
Comment (matleppa) [inline apinf_packages/apis/server/api.js]: Also parameter "statusList" is needed here in order to be visible and usable in Swagger UI.
To get an example of similar parameter, see implementation of parameter "managedAPIs" in GET /apis method.

File: apinf_packages/apis/server/api.js [hunk 2]
```diff
@@ -1921,5 +1923,120 @@ CatalogV1.addRoute('apis/:id/proxyBackend', {
...
+// Request /rest/v1/apis/:id/monitoring/
+CatalogV1.addRoute('apis/:id/monitoring', {
+  get: {
+    authRequired: false,
+    swagger: {
+      tags: [
+        CatalogV1.swagger.tags.api,
+      ],
+      summary: 'Get latest Monitoring status of API',
+      description: 'In return user get object with latest API code status',
```
Comment (matleppa) [inline apinf_packages/apis/server/api.js]: In return user get object…
-->
In return user gets object...

File: apinf_packages/apis/server/api.js [hunk 3]
```diff
@@ -1921,5 +1923,120 @@ CatalogV1.addRoute('apis/:id/proxyBackend', {
...
+        // User have admin rights
+        // Get Manager ID from header
+        const managerId = this.request.headers['x-user-id'];
+         // Check if requestor is administrator
+        const requestorIsAdmin = Roles.userIsInRole(managerId, ['admin']);
+        if (!requestorIsAdmin) {
+          return errorMessagePayload(403, 'User does not have permission');
+        }
+         // Get list of monitoring statuses
+        monitoringStatusList = MonitoringData.find(apiId, options).fetch();
```
Comment (matleppa) [inline apinf_packages/apis/server/api.js]: 1. Curly braces are needed in order to get result
monitoringStatusList = MonitoringData.find(apiId, options).fetch();
-->
monitoringStatusList = MonitoringData.find({ apiId }, options).fetch();

2. Now complete monitoringData object is set to monitoringStatusList, however, what we need is only last 24 hours of response records, so filter those and add only them into monitoringStatusList.

Comment (55) [reply apinf_packages/apis/server/api.js]: @matleppa about 2 I remember we discussed that it should come from application functionality not API. Right?

Comment (matleppa) [reply apinf_packages/apis/server/api.js]: I think that final solution would be to add a search parameter for period length into the request to indicate the amount of expected monitor statuses. 
That way the API does not do decisions of its own, but the application has the control and API returns the data which was asked, not more.

I'm not aware of if you have agreed about some certain amount of this work with Chandra/Ilari, so concerning item 2 either way is OK for me:
- return list of found monitor statuses
- implement a parameter to restrict the returned list of monitored statuses

**Pull request merged**